### PR TITLE
RRD errors should be a bad remember

### PR DESCRIPTION
**Pull request by bouda1**
## Description

REFS: MON-150015

### CODE REVIEW

**Review by sechkem — APPROVED** [review]

File: broker/bam/src/bool_service.cc [hunk 1]
```diff
@@ -167,6 +167,31 @@ void bool_service::service_update(
...
+ */
+void bool_service::service_update(
+    const std::shared_ptr<neb::pb_adaptive_service_status>& status,
+    io::stream* visitor) {
+  auto& o = status->obj();
+  SPDLOG_LOGGER_TRACE(_logger,
+                      "bool_service: service ({},{}) updated with "
+                      "neb::pb_adaptive_service_status downtime: {}",
+                      o.host_id(), o.service_id(),
+                      o.scheduled_downtime_depth());
```
Comment (jean-christophe81) [inline broker/bam/src/bool_service.cc:184]: it's an optional, what happens if not set?

File: broker/bam/src/bool_service.cc [hunk 2]
```diff
@@ -167,6 +167,31 @@ void bool_service::service_update(
...
+  if (o.host_id() == _host_id && o.service_id() == _service_id) {
+    bool new_in_downtime = o.scheduled_downtime_depth() > 0;
```
Comment (jean-christophe81) [inline broker/bam/src/bool_service.cc:186]: same

File: broker/lua/inc/com/centreon/broker/lua/macro_cache.hh [hunk 3]
```diff
@@ -42,7 +41,7 @@ namespace com::centreon::broker::lua {
 class macro_cache {
   std::shared_ptr<persistent_cache> _cache;
   absl::flat_hash_map<uint64_t, std::shared_ptr<io::data>> _instances;
-  absl::flat_hash_map<uint64_t, std::shared_ptr<io::data>> _hosts;
+  absl::flat_hash_map<uint64_t, std::shared_ptr<neb::pb_host>> _hosts;
```
Comment (jean-christophe81) [inline broker/lua/inc/com/centreon/broker/lua/macro_cache.hh:44]: how manage the case of neb::host objects are yet in the cache before upgrade?

File: broker/lua/inc/com/centreon/broker/lua/macro_cache.hh [hunk 4]
```diff
@@ -54,7 +53,8 @@ class macro_cache {
       _host_group_members;
   absl::flat_hash_map<std::pair<uint64_t, uint64_t>, std::shared_ptr<io::data>>
       _custom_vars;
-  absl::flat_hash_map<std::pair<uint64_t, uint64_t>, std::shared_ptr<io::data>>
+  absl::flat_hash_map<std::pair<uint64_t, uint64_t>,
```
Comment (jean-christophe81) [inline broker/lua/inc/com/centreon/broker/lua/macro_cache.hh:56]: same

File: broker/lua/src/macro_cache.cc [hunk 5]
```diff
@@ -975,11 +1105,42 @@ void macro_cache::_process_pb_service(std::shared_ptr<io::data> const& data) {
...
+        obj.host_id(), obj.service_id());
+    return;
+  }
+
+  auto& svc = std::static_pointer_cast<neb::pb_service>(it->second)->mut_obj();
+  if (obj.has_acknowledgement_type())
+    svc.set_acknowledgement_type(obj.acknowledgement_type());
+  if (obj.has_scheduled_downtime_depth())
+    svc.set_scheduled_downtime_depth(obj.scheduled_downtime_depth());
+}
```
Comment (jean-christophe81) [inline broker/lua/src/macro_cache.cc:1142]: has_notification_number ?

File: broker/neb/src/callbacks.cc [hunk 6]
```diff
@@ -18,36 +18,25 @@
 
 #include "com/centreon/broker/neb/callbacks.hh"
 
+#include <absl/strings/numbers.h>
 #include <absl/strings/str_split.h>
 #include <unistd.h>
-
-#include "bbdo/neb.pb.h"
-#include "opentelemetry/proto/collector/metrics/v1/metrics_service.pb.h"
+#include <string_view>
```
Comment (jean-christophe81) [inline broker/neb/src/callbacks.cc:24]: why not in precomp?

File: common/vault/vault_access.cc [hunk 7]
```diff
@@ -20,6 +20,7 @@
 #include <absl/strings/str_split.h>
 #include <exception>
```
Comment (jean-christophe81) [inline common/vault/vault_access.cc:21]: to remove?

File: broker/neb/src/callbacks.cc [hunk 8]
```diff
@@ -2121,90 +2109,115 @@ int neb::callback_host_status(int callback_type, void* data) {
...
-      static_cast<nebstruct_host_status_data*>(data)->object_ptr)};
+  nebstruct_host_status_data* hsd =
+      static_cast<nebstruct_host_status_data*>(data);
+  const engine::host* eh = static_cast<engine::host*>(hsd->object_ptr);
+
+  auto handle_acknowledgement = [](uint16_t state, auto& hscr) {
+    auto it = gl_acknowledgements.find(std::make_pair(hscr.host_id(), 0u));
+    if (it != gl_acknowledgements.end() &&
+        hscr.acknowledgement_type() == AckType::NONE) {
+      if (it->second->type() == make_type(io::neb, de_pb_acknowledgement)) {
```
Comment (jean-christophe81) [inline broker/neb/src/callbacks.cc:2127]: can we ahve another thing that pb_ack?

File: broker/neb/src/callbacks.cc [hunk 9]
```diff
@@ -3386,140 +3426,132 @@ int32_t neb::callback_pb_service_status(int callback_type [[maybe_unused]],
...
+  auto handle_acknowledgement = [](uint16_t state, auto& r) {
+    neb_logger->debug("Looking for acknowledgement on service ({}:{})",
+                      r.host_id(), r.service_id());
+    auto it =
+        gl_acknowledgements.find(std::make_pair(r.host_id(), r.service_id()));
+    if (it != gl_acknowledgements.end() &&
+        r.acknowledgement_type() == AckType::NONE) {
+      neb_logger->debug("acknowledgement found on service ({}:{})", r.host_id(),
+                        r.service_id());
+      if (it->second->type() == make_type(io::neb, de_pb_acknowledgement)) {
```
Comment (jean-christophe81) [inline broker/neb/src/callbacks.cc:3447]: if mandatory?

**Review by jean-christophe81 — APPROVED** [review]

**Pull request merged**